vector-im/element-x-android
# [Room Details] Block & unblock user

## What

Adds shared UI and logic code to block and unblock users from both room details screen and member details screen.

## Why

Implements #339 .

## Tests

You'll need to test this both in a DM room and a regular room.

1. Select a user you want to ignore.
2. Open the DM or regular room. Make sure it contains messages from that user (other events, including undecrypted, ones don't count).
3. Go to the room details. If you're on a regular room navigate to the member details screen for the user you chose before.
4. Block the user, come back to the timeline.
5. Their messages should have disappeared from the timeline. In the DM screen the composer shouldn't be present either.
6. Go again to the user's details, unblock them.
7. The messages should be back in the timeline, the composer should be visible again.

## Video

[block_unblock.webm](https://user-images.githubusercontent.com/480955/233005413-d19f70db-5237-4f2f-81ed-3b465d83c418.webm)
